web/src/components/TaskCtaArea.tsx

```diff
@@ -0,0 +1,123 @@
+import { useRouter } from "next/router";
+import React, { ReactElement } from "react";
+
+interface ButtonsArrayElement {
+  text: string;
+  destination: string;
+  onClick: () => void;
+}
+
+const TaskFooter = (props: React.ComponentPropsWithoutRef<"div">): JSX.Element => {
+  const { children, ...attributes } = props;
+  return (
+    <div
+      data-testid="cta-area"
+      style={{ marginTop: "2rem" }}
+      className="flex flex-justify-end bg-base-lightest margin-x-neg-4 padding-3 margin-bottom-neg-4 flex-column mobile-lg:flex-row radius-bottom-lg"
+      {...attributes}
+    >
+      <div className="mobile-lg:margin-bottom-0 margin-bottom-1">{children}</div>
+    </div>
+  );
+};
+
+interface Props {
+  postOnboardingQuestion: PostOnboarding | undefined;
+  task?: Task | undefined;
+  forceDisplay?: boolean | undefined;
+  onboardingKey?: keyof ProfileData | undefined;
+}
+
+export const TaskCtaArea = (props: Props): ReactElement => {
+  const ctaButtons: ButtonsArrayElement[] = [];
+  const { business } = useUserData();
+  const router = useRouter();
+
+  const taskCtaDestination = props.task?.callToActionLink;
+  let taskCtaText = props.task?.callToActionText ?? "";
+
+  if (taskCtaText && taskCtaText.length === 0) {
+    taskCtaText = "Start Application";
+  }
+
+  const link1 = props.postOnboardingQuestion?.callToActionYesLink1;
+  const text1 = props.postOnboardingQuestion?.callToActionYesText1;
+  const link2 = props.postOnboardingQuestion?.callToActionYesLink2;
+  const text2 = props.postOnboardingQuestion?.callToActionYesText2;
+
+  const hasTaskCta = taskCtaDestination;
+  const hasPostOnboarding = !!props.postOnboardingQuestion;
+
+  const hasAnsweredYesToPostOnboarding =
+    (props.onboardingKey && business?.profileData[props.onboardingKey] === true) || props.forceDisplay;
+
+  if (hasPostOnboarding && hasAnsweredYesToPostOnboarding) {
+    if (link1 && link1.length > 0 && text1 && text1.length > 0) {
+      ctaButtons.push({
+        text: text1,
+        destination: link1,
+        onClick: () => {
+          if (props.forceDisplay) {
+            openInNewTab(link1);
+          } else {
+            router.push(link1);
+          }
+        },
+      });
+    }
+    if (link2 && link2.length > 0 && text2 && text2.length > 0) {
+      ctaButtons.push({
+        text: text2,
+        destination: link2,
+        onClick: () => {
+          if (props.forceDisplay) {
+            openInNewTab(link2);
+          } else {
+            router.push(link2);
+          }
+        },
+      });
+    }
+  } else {
+    if (hasTaskCta) {
+      ctaButtons.push({
+        text: taskCtaText,
+        destination: taskCtaDestination,
+        onClick: () => {
+          if (props.forceDisplay) {
+            openInNewTab(taskCtaDestination);
+          } else {
+            router.push(taskCtaDestination);
+          }
+        },
+      });
+    }
+  }
+
+  if (ctaButtons.length === 1) {
+    const [buttonsMeta] = ctaButtons;
```

Comment:
[sand] This is not my favorite variable name, because it's not very descriptive. Maybe something to make it clear at quick glance that this is the one (and only) button we care about? idk, `singularButton` or something like that? Idk